wsmaster/codenvy-hosted-api-license/src/main/java/com/codenvy/api/license/server/CodenvyLicenseManager.java

```diff
@@ -151,34 +117,33 @@ private void removeActionsOfExpiredLicense() throws ServerException {
      */
     @Nullable
     public CodenvyLicense load() throws LicenseException {
-        String licenseText = readLicenseText();
-        if (isNullOrEmpty(licenseText)) {
-            throw new LicenseNotFoundException("Codenvy license not found");
-        }
-
-        return licenseFactory.create(licenseText);
+        String licenseText = codenvyLicenseStorage.loadLicense();
+        CodenvyLicense codenvyLicense = licenseFactory.create(licenseText);
+        codenvyLicenseActivator.validateActivation(codenvyLicense);
+        return codenvyLicense;
     }
 
     /**
      * Deletes Codenvy license from the storage.
      *
+     * @throws LicenseNotFoundException
+     *      if Codenvy license not found
      * @throws LicenseException
-     *         if error occurred while deleting license
+     *       if error occurred while deleting license
      */
     public void delete() throws LicenseException, ApiException {
-        String licenseText = readLicenseText();
-        String licenseQualifier = extractLicenseId(licenseText);
+        String licenseText = codenvyLicenseStorage.loadLicense();
 
         try {
-            Files.delete(licenseFile);
-        } catch (NoSuchFileException e) {
-            throw new LicenseNotFoundException("Codenvy license not found");
-        } catch (IOException e) {
-            throw new LicenseException(e.getMessage(), e);
-        }
+            CodenvyLicense codenvyLicense = licenseFactory.create(licenseText);
+            codenvyLicenseStorage.clean();
 
-        codenvyLicenseActionDao.remove(PRODUCT_LICENSE, EXPIRED);
-        addLicenseAction(PRODUCT_LICENSE, EXPIRED, licenseQualifier);
+            for (CodenvyLicenseManagerObserver observer : observers) {
+                observer.onProductLicenseDeleted(codenvyLicense);
+            }
+        } catch (InvalidLicenseException e) {
+            codenvyLicenseStorage.clean();
```

Comment:
Why do we need to call clean() method again in case of exception?
IMHO it's better not to remove existed files in case of error to make it possible to use existed license to investigate the problem.